_posts/2018-11-06-simple-slackbot.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title:  "Simple Slackbot"
+date: 2018-11-06 00:03:30 +0200
+categories: javascript
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2018-11-06-simple-slackbot/poster.jpg
+---
+
+Recently I've built a small slackbot that is posting with certain regularity in specified slack-channel. I will tell you about the idea behind it and also share some technical details.
+
+
+## The Idea
+
+The idea to build such a bot came from one colleague of mine - [Jacek Smolak][jacek]. At that point he was reading the book ["97 Things Every Programmer Should Know"][book-link]. In order to share those programmer tips with others and also remind them to himself, he came up with idea of posting each of them one by one on a daily basis in appropriate slack channel. For us this slack channel is #craft, you probably have something similar amoung your channels in slack.
+
+
+## Implementation
+
+The implementation is dead simple. As it was written with JavaScript, the easiest way was to use an official [slack npm package][slack-npm]. Posting something in specific channel becoming a one-liner that returns a Promise that we can handle depending if it is success or failure:
```

Comment:
... written **in** ...

Reply:
`One-liner` is referring to a code that send the message. I've still rewrote that sentence to make it more understandable. 
I don't use here the `Promise` directly, but the slack library does, so I've just mentioned that from this call you get do "then" or "catch" as its a Promise